examples/coap/src/bin/coapserver-coaphandler.rs

```diff
@@ -32,7 +32,7 @@ struct EdhocHandler {
 ///
 /// Places using this function may be simplified if From/Into is specified (possibly after
 /// enlarging the Error type)
-fn too_small(_e: MessageBufferError) -> Error {
+fn too_small(_e: EdhocBufferError) -> Error {
```

Comment:
This is the only public fall-out of the change -- and even that I'm changing here merely to avoid the deprecation warning.

In terms of semver compatibility, this is all a real breaking change, b/c two types that were previously distinct are now one, and that is visible for whoever impl'd any own trait on both MessageBufferError and EdhocBufferError (or the respective non-error types).